engagement_database/engagement_database.py

```diff
@@ -275,6 +275,25 @@ def delete_doc(self, path, transaction=None):
         else:
             transaction.delete(ref)
 
+    def delete_message_and_history(self, message_id, transaction=None):
+        """
+        Deletes a message and its history entries.
+
+        :param message_id: Id of message to delete.
+        :type message_id: str
+        :param transaction: Transaction to run this update in or None.
+                            If None, writes immediately, otherwise adds the updates to a transaction that will need
+                            to be explicitly committed elsewhere.
+        :type transaction: google.cloud.firestore.Transaction | None
+        """
+        message_snapshot = self._message_ref(message_id).get()
```

Comment:
I think this needs to be transactional even if this is not taking place in an externally-managed transaction. This is to ensure that in no circumstances can we delete a message without deleting all of its history too. As an example, you can look at `set_message` in this file, whose implementation ensures that we can't update a message without updating its history.

Reply:
Updated, thanks.